src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -0,0 +1,375 @@
+/*******************************************************************\
+
+Module: Symbol Analyzer
+
+Author: Malte Mues <mail.mues@gmail.com>
+        Daniel Poetzl
+
+\*******************************************************************/
+
+#include "analyze_symbol.h"
+
+#include <util/c_types.h>
+#include <util/c_types_util.h>
+#include <util/config.h>
+#include <util/expr_initializer.h>
+#include <util/string_constant.h>
+
+gdb_value_extractort::gdb_value_extractort(
+  const symbol_tablet &symbol_table,
+  const char *binary)
+  : gdb_api(binary),
+    symbol_table(symbol_table),
+    ns(symbol_table),
+    c_converter(ns, expr2c_configurationt::clean_configuration),
+    allocate_objects(ID_C, source_locationt(), "", this->symbol_table)
+{
+}
+
+void gdb_value_extractort::analyze_symbols(
+  const std::vector<std::string> &symbols)
+{
+  // record addresses of given symbols
+  for(const auto &id : symbols)
+  {
+    const symbol_exprt symbol_expr(id, typet());
```

Comment:
I'd suggest to use `symbol_exprt::typeless`. Though, reading a bit further: maybe actually `const symbol_exprt symbol_expr = ns.lookup(id).symbol_expr();`?

Reply:
Opted for the latter. This and the remaining changes within `analyze_symbol.cpp` are in https://github.com/danpoe/cbmc/pull/9/commits/339f37c23a4583ee758a8570c1d8d5247e49b90d.